app/models/local_photo.rb

```diff
@@ -84,7 +88,76 @@ class LocalPhoto < Photo
     "SAMSUNG CAMERA PICTURES",
     "MINOLTA DIGITAL CAMERA"
   ]
-  
+
+  def self.s3_host_alias( public = false )
+    unless ( CONFIG.s3_public_host || CONFIG.s3_public_bucket )
+      return ( CONFIG.s3_host || CONFIG.s3_bucket )
+    end
+    public ?
+      ( CONFIG.s3_public_host || CONFIG.s3_public_bucket ) :
+      ( CONFIG.s3_host || CONFIG.s3_bucket )
+  end
+
+  def self.s3_region( public = false )
+    return CONFIG.s3_region unless CONFIG.s3_public_region
+    public ? CONFIG.s3_public_region : CONFIG.s3_region
+  end
+
+  def self.s3_bucket( public = false )
+    return CONFIG.s3_bucket unless CONFIG.s3_public_bucket
+    public ? CONFIG.s3_public_bucket : CONFIG.s3_bucket
+  end
+
+  def self.s3_permissions( public = false )
+    return "public-read" unless CONFIG.s3_public_bucket && CONFIG.s3_public_acl
+    public ? CONFIG.s3_public_acl : "public-read"
+  end
+
+  def s3_host_alias
+    LocalPhoto.s3_host_alias( self.s3_account )
+  end
+
+  def s3_region
+    LocalPhoto.s3_region( self.s3_account )
+  end
+
+  def s3_bucket
+    LocalPhoto.s3_bucket( self.s3_account )
+  end
+
+  def s3_permissions
+    LocalPhoto.s3_permissions( self.s3_account )
+  end
+
+  def set_s3_account
+    self.s3_account = self.is_public? ? "public" : nil
+  end
+
+  def is_public?
```

Comment:
I know I'm picking nits, but all iNat photos are public. Could we use `is_s3_public?` instead for clearer semantics (if more awkward typing)? Alternatives might be `odp_candidate?` or `odp_eligible?`

Reply:
I made a change, but I'm open to other changes. I have used "public" a bunch, such as in the s3 config parameters, which maybe isn't the best. I'm open to any naming changes. We could go with odp across the board if we like that better

Reply:
Thanks. I think `is_in_public_s3_bucket?` expresses its meaning more clearly, so that works for me. FWIW, I think it's usually preferable to avoid verbs in boolean methods like this. It looks the same in conditionals (e.g. `if in_public_s3_bucket?`) and better in rspec (e.g. `expect( photo ).to be_in_public_s3_bucket?`)